.github/workflows/0-on-demand.yaml

```diff
@@ -0,0 +1,94 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+#
+name: On Demand Tests
+
+on:
+  workflow_dispatch:
+    inputs:
+      enable_ngrok_debug:
```

Comment:
What is the expected behavior for contributors to enable this debugging?
Are they supposed to set up a secret in their forked repo, enable this input, and run tests in their repo?

Since the secret is not configured in the upstream repo, no debugging would be enabled in the upstream repo, right?

Reply:
The NGROK support is now only in the "on demand" task that must be manually activated.
So you trigger it when you want. Works in your fork, so you have to add the token only for your it. When it is enabled in the actions a menu to trigger manually appears, and you have to check the box and select the test you want to run as below.

![Screenshot 2022-12-15 at 06 40 26](https://user-images.githubusercontent.com/1595312/207790563-c5d85b79-13c6-4273-8f8e-6203dedff050.png)